src/azul/service/repository_controller.py

```diff
@@ -61,7 +64,10 @@ def download_file(self,
             file_version = file['version']
             drs_path = file['drs_path']
             if file_name is None:
-                file_name = file['name']
+                # To prevent file overwrites due to files with the same name
+                # we insert the crc32c value of the file into the filename.
+                base_name, extension = file_name_partition(file['name'])
+                file_name = base_name + '_' + file['crc32c'] + '.' + extension
```

Comment:
if the extension is the empty string `''` then there will be a `.` at the end of the filename is that expected?

Reply:
Good catch. Will fix.